src/core/lib/layer.js

```diff
@@ -610,29 +616,41 @@ export default class Layer {
   }
 
   _updateBaseUniforms() {
-    this.setUniforms({
```

Comment:
Is there a reason to not call `setUniforms` here?

Reply:
> Is there a reason to not call setUniforms here?

Having a `setUniforms` method on the layer has been pointed out as point of confusion when writing new layers... should the layer call, or override `Layer.setUniforms`, or call `model.setUniforms`, or even `Program.setUniforms`, or should I use `Layer.draw(uniforms)` etc. 

In my mind, `Layer.setUniforms` is not a well defined function as it stands (when should it be called? what uniforms does it handle?). It does not have an underscore so it looks like a public method, but it is not documented as far as I can tell. So this was intended as a first step towards phasing it out.